plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/PyStringCodeCompletion.java

```diff
@@ -240,6 +245,87 @@ public TokensOrProposalsList getCodeCompletionProposals(CompletionRequest reques
         return ret;
     }
 
+    private TokensOrProposalsList getCompletionsForTypedDict(CompletionRequest request) throws CoreException,
```

Comment:
I think that a `BadLocationException` should be just logged in this function (and nothing should be returned) instead of throwing an error.